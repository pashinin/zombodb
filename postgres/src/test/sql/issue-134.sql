SET datestyle TO 'iso, mdy';
CREATE TABLE multifield AS SELECT * FROM so_users ORDER BY id LIMIT 10;
SELECT zdb_define_mapping('multifield', 'display_name', '{
    "type": "string", "analyzer": "phrase",
    "fields": {
        "raw":   { "type": "string", "index": "not_analyzed" }
    }
}');
CREATE INDEX idxmultifield ON multifield USING zombodb (zdb('multifield', ctid), zdb(multifield)) WITH (url='http://localhost:9200/');

SELECT display_name FROM multifield WHERE zdb('multifield', ctid) ==> 'display_name:robert';

SELECT display_name FROM multifield WHERE zdb('multifield', ctid) ==> 'display_name.raw:robert';
SELECT display_name FROM multifield WHERE zdb('multifield', ctid) ==> 'display_name.raw:"Robert Cartaino"';

DROP TABLE multifield;