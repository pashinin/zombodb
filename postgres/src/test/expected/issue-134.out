SET datestyle TO 'iso, mdy';
CREATE TABLE multifield AS SELECT * FROM so_users ORDER BY id LIMIT 10;
SELECT zdb_define_mapping('multifield', 'display_name', '{
    "type": "string", "analyzer": "phrase",
    "fields": {
        "raw":   { "type": "string", "index": "not_analyzed" }
    }
}');
 zdb_define_mapping 
--------------------
 
(1 row)

CREATE INDEX idxmultifield ON multifield USING zombodb (zdb('multifield', ctid), zdb(multifield)) WITH (url='http://localhost:9200/');
WARNING:  No primary key detected for public.multifield, continuing anyway
SELECT display_name FROM multifield WHERE zdb('multifield', ctid) ==> 'display_name:robert';
  display_name   
-----------------
 Robert Cartaino
(1 row)

SELECT display_name FROM multifield WHERE zdb('multifield', ctid) ==> 'display_name.raw:robert';
 display_name 
--------------
(0 rows)

SELECT display_name FROM multifield WHERE zdb('multifield', ctid) ==> 'display_name.raw:"Robert Cartaino"';
  display_name   
-----------------
 Robert Cartaino
(1 row)

DROP TABLE multifield;
